### PR TITLE
Marian

### PR DESCRIPTION
**Pull request by marianmagur**

### CODE REVIEW

File: homeworks/magurmarian/hw01/calculation.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                step.passx(f'num1:{num1} ,num2:{num2}',
+                           from_exception=ZeroDivisionError('ZeroDivisionError'))
+            except AssertionError:
+                self.skipped('Result is less then 0)')
+
+
+def inputs():
+    parser = argparse.ArgumentParser(description="standalone parser")
+    parser.add_argument('-num1', dest='num1', type=int, required=False, default=parameters['num1'])
+    parser.add_argument('-num2', dest='num2', type=int, required=False, default=parameters['num1'])
```
Comment (Tempos) [inline homeworks/magurmarian/hw01/calculation.py]: Typo
```suggestion
    parser.add_argument('-num2', dest='num2', type=int, required=False, default=parameters['num2'])
```

File: homeworks/magurmarian/hw01/somefunc.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+def add(num1, num2):
```
Comment (Tempos) [inline homeworks/magurmarian/hw01/somefunc.py]: - This file was supposed to have the "**calculation**.py" name.
- doc-strings are missing

File: homeworks/magurmarian/hw01/calculation.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @test
+    def test_add(self, steps, num1, num2):
+        with steps.start('Result bigger then 0') as step:
+            if add(num1, num2) > 0:
+                assert add(num1, num2) > 0
+            else:
+                self.skipped('Result is less then 0)')
+
+    @test
+    def test_devide(self, steps, num1, num2):
```
Comment (Tempos) [inline homeworks/magurmarian/hw01/calculation.py]: Possibly a typo in function name.
```suggestion
    def test_divide(self, steps, num1, num2):
```

File: homeworks/magurmarian/hw01/calculation.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    'num2': 0
+}
+
+
+class SmokeTest(Testcase):
+
+    @test
+    def test_add(self, steps, num1, num2):
+        with steps.start('Result bigger then 0') as step:
+            if add(num1, num2) > 0:
```
Comment (Tempos) [inline homeworks/magurmarian/hw01/calculation.py]: Tests should be skipped if result is less than zero. In your case, zero is not included.
Also, steps comments could me more informative.

File: homeworks/magurmarian/hw01/test_calculation.py [hunk 5]
```diff
@@ -11,17 +12,18 @@
...
     @test
-    def test_add(self, steps, num1, num2):
-        with steps.start('Result bigger then 0') as step:
-            if add(num1, num2) > 0:
-                assert add(num1, num2) > 0
-            else:
-                self.skipped('Result is less then 0)')
+    def test_add(self, num1, num2):
+        """method for testing add function"""
+        if add(num1, num2) >= 0:
```
Comment (Tempos) [inline homeworks/magurmarian/hw01/test_calculation.py]: Excessive code. Simply skipping, when function results in negative value, will be enough.

**Comment by extsoft** [comment]
@marianmagur @Tempos next time please care about adding me as a reviewer

File: homeworks/magurmarian/hw01/test_calculation.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    @test
+    def test_add(self, num1, num2):
+        """method for testing add function"""
+        if add(num1, num2) >= 0:
+            assert add(num1, num2) >= 0
+        else:
+            self.skipped('Result is less then 0)')
+
+    @test
+    def test_devide(self, steps, num1, num2):
```
Comment (extsoft) [inline homeworks/magurmarian/hw01/test_calculation.py]: The test has to be skipped if result is less then 0.

File: homeworks/magurmarian/hw01/test_calculation.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                step.passx(f'num1:{num1} ,num2:{num2}',
+                           from_exception=ZeroDivisionError('ZeroDivisionError'))
+            except AssertionError:
+                self.skipped('Result is less then 0)')
+
+
+def inputs():
+    """parsing user arguments here"""
+    parser = argparse.ArgumentParser(description="standalone parser")
+    parser.add_argument('-num1', dest='num1', type=int, required=False, default=parameters['num1'])
```
Comment (extsoft) [inline homeworks/magurmarian/hw01/test_calculation.py]: @marianmagur do you really need to work with `parameters`? Can you simplify the implementation?

Comment (marianmagur) [reply homeworks/magurmarian/hw01/test_calculation.py]: Done!

**Pull request merged**